rust/src/router/producer.rs

```diff
@@ -106,14 +106,14 @@ pub struct RtpStreamRecv {
 }
 
 impl RtpStreamRecv {
-    pub(crate) fn from_fbs(dump: &rtp_stream::Dump) -> Self {
-        Self {
-            params: RtpStreamParams::from_fbs(&dump.params),
-            score: dump.score,
-            rtx_stream: dump.rtx_stream.clone().map(|stream| RtxStream {
-                params: RtxStreamParams::from_fbs(&stream.params),
+    pub(crate) fn from_fbs_ref(dump: rtp_stream::DumpRef<'_>) -> Result<Self, Box<dyn Error>> {
+        Ok(Self {
+            params: RtpStreamParams::from_fbs_ref(dump.params()?)?,
+            score: dump.score()?,
+            rtx_stream: dump.rtx_stream()?.map(|stream| RtxStream {
+                params: RtxStreamParams::from_fbs_ref(stream.params().unwrap()).unwrap(),
```

Comment:
@nazar-pc, is there any way to avoid calling those `unwrap` and use the `?` here instead? I couldn't find it as the `map()` closure does not return a `Result` or `Option`.

Reply:
Unless I miss something, `map()` in any language constructs a new array so it's impossible that it returns a reference (a reference to what? it's returning something that it has internally created).

Reply:
Why not? There is [`and_then`](https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.and_then) and [`transpose`](https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.transpose) depending on what you want to do here.

Reply:
I can't get rid of those.., will retry tomorrow or ask for help if I can't.

Reply:
Done